nextcord/scheduled_events.py

```diff
@@ -0,0 +1,388 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Tuple, TYPE_CHECKING
+
+from .enums import ScheduledEventPrivacyLevel
+from .iterators import ScheduledEventUserIterator
+from .mixins import Hashable
+from .types.snowflake import Snowflake
+from .utils import MISSING, parse_time
+
+__all__: Tuple[str] = (
+    'EntityMetadata',
+    'ScheduledEventUser',
+    'ScheduledEvent'
+)
+
+if TYPE_CHECKING:
+    from datetime import datetime
+
+    from .abc import GuildChannel
+    from .enums import ScheduledEventStatus, ScheduledEventEntityType
+    from .guild import Guild
+    from .member import Member
+    from .state import ConnectionState
+    from .types.scheduled_events import (
+        ScheduledEvent as ScheduledEventPayload,
+        ScheduledEventUser as ScheduledEventUserPayload
+    )
+    from .user import User
+
+
+class EntityMetadata:
+    """Represents the metadata for an event
+
+    Parameters
+    ----------
+    location : Optional[str]
+        The location of the event, defaults to None
+    """
+
+    # kwargs is for if more attributes are added to the api, prevent breakage
+    def __init__(self, *, location: Optional[str] = None, **kwargs: Any) -> None:
+        self.location: Optional[str] = location
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+
+class ScheduledEventUser(Hashable):
+    """Represents a user in a scheduled event
+
+    Attributes
+    ----------
+    event: :class:`ScheduledEvent`
+        The event the user is interested in.
+    user: :class:`User`
+        The related user object.
+    member: Optional[:class:`Member`]
+        The related member object, if requested with 
+        :meth:`ScheduledEvent.fetch_users`.
+    """
+    __slots__: Tuple[str] = (
+        'event',
+        'user',
+        'member'
+    )
+
+    def __init__(
+        self,
+        *,
+        event: ScheduledEvent,
+        state: ConnectionState,
+        data: ScheduledEventUserPayload
+    ) -> None:
+        self.event = event
+        self._state = state
+
+        self._update(data)
+
+    def _update(self, data: ScheduledEventUserPayload) -> None:
+        self.user: User = self._state.store_user(data['user'])
+        if member := data.get('member'):
+            if not self._state.member_cache_flags._empty:
+                try:
+                    self.member: Optional[Member] = self.guild.get_member(
+                        member['id']
+                    )
+                except KeyError:
+                    m = Member(data=member, guild=self.guild, state=self._state)
+                    self.member: Optional[Member] = m
+            else:
+                m = Member(data=member, guild=self.guild, state=self._state)
+                self.member: Optional[Member] = m
```

Comment:
Is there a need for making this 2 lines rather then 1?

Reply:
i think it takes up too much line length